pkg/codegen/go/gen_program.go

```diff
@@ -354,9 +354,14 @@ func (g *generator) collectImports(
 	for _, n := range program.Nodes {
 		if r, isResource := n.(*pcl.Resource); isResource {
 			pkg, mod, name, _ := r.DecomposeToken()
-			if pkg == "pulumi" && mod == "providers" {
-				pkg = name
-				mod = ""
+			if pkg == "pulumi" {
+				if mod == "providers" {
+					pkg = name
+					mod = ""
+				} else if mod == "pulumi" {
```

Comment:
Where would `pulumi:pulumi:*` be a valid token?

I'll definitely want to see a test for this PR.

Reply:
Stack references are "pulumi:pulumi:StackReference"